sympy/solvers/ode/single.py

```diff
@@ -2482,12 +2482,17 @@ def _undetermined_coefficients_match(self,expr, x, func=None, eq_homogeneous=S.Z
         Examples
         ========
 
-        >>> from sympy import log, exp
-        >>> from sympy.solvers.ode.ode import _undetermined_coefficients_match
+        >>> from sympy import log, exp, Function
+        >>> from sympy.solvers.ode.single import NthLinearConstantCoeffUndeterminedCoefficients,SingleODEProblem
         >>> from sympy.abc import x
-        >>> _undetermined_coefficients_match(9*x*exp(x) + exp(-x), x)
+        >>> f = Function('f')
+        >>> eq = 9*x*exp(x) + exp(-x)
+        >>> obj = NthLinearConstantCoeffUndeterminedCoefficients(SingleODEProblem(eq,f(x),x))
+        >>> obj._undetermined_coefficients_match(eq,x)
         {'test': True, 'trialset': {x*exp(x), exp(-x), exp(x)}}
-        >>> _undetermined_coefficients_match(log(x), x)
+        >>> eq = log(x)
+        >>> obj = NthLinearConstantCoeffUndeterminedCoefficients(SingleODEProblem(eq,f(x),x))
+        >>> obj._undetermined_coefficients_match(eq,x)
         {'test': False}
```

Comment:
It's probably better to give an example using `dsolve` here.

Reply:
This docstring is just for getting trial set method. For examples of undet_coeff solver the docstring at class level has examples with dsolve.